src/uu/wc/src/wc.rs

```diff
@@ -53,17 +70,92 @@ impl Settings {
             show_max_line_length: false,
         }
     }
+
+    fn number_enabled(&self) -> u32 {
+        let mut result = 0;
+        result += self.show_bytes as u32;
+        result += self.show_chars as u32;
+        result += self.show_lines as u32;
+        result += self.show_max_line_length as u32;
+        result += self.show_words as u32;
+        result
+    }
+}
+
+#[cfg(unix)]
+trait WordCountable: AsRawFd + Read {
+    type Buffered: BufRead;
+    fn get_buffered(self) -> Self::Buffered;
 }
+#[cfg(not(unix))]
+trait WordCountable: Read {
+    type Buffered: BufRead;
+    fn get_buffered(self) -> Self::Buffered;
+}
+
+impl WordCountable for StdinLock<'_> {
+    type Buffered = Self;
 
-struct Result {
-    title: String,
+    fn get_buffered(self) -> Self::Buffered {
+        return self;
+    }
+}
+impl WordCountable for File {
+    type Buffered = BufReader<Self>;
+
+    fn get_buffered(self) -> Self::Buffered {
+        return BufReader::new(self);
```

Comment:
`return` is unnecessary.  I think `rustfmt` might automatically handle this for you.

Reply:
Right. I've been writing too much C++ lately :(